apps/studio/components/interfaces/Realtime/RealtimeSettings.tsx

```diff
@@ -0,0 +1,345 @@
+                      header={
+                        <FormSectionLabel
+                          description={
+                            <p className="text-foreground-lighter text-sm !mt-1">
+                              Sets maximum number of concurrent users rate limit
+                            </p>
+                          }
+                        >
+                          Max concurrent users
+                        </FormSectionLabel>
+                      }
+                    >
+                      <FormSectionContent loading={isLoading}>
+                        <FormControl_Shadcn_>
+                          <Input_Shadcn_
+                            {...field}
+                            type="number"
+                            disabled={!canUpdateConfig}
+                            value={field.value || ''}
+                            {...form.register('max_concurrent_users', { valueAsNumber: true })}
+                          />
+                        </FormControl_Shadcn_>
+                      </FormSectionContent>
+                    </FormSection>
+                  )}
+                />
+              </CardContent>
+              <CardContent>
+                <FormField_Shadcn_
+                  control={form.control}
+                  name="max_events_per_second"
+                  render={({ field }) => (
+                    <FormSection
+                      className="!p-0 !py-2"
+                      header={
+                        <FormSectionLabel
+                          description={
+                            <p className="text-foreground-lighter text-sm !mt-1">
+                              Sets maximum number of events per second rate per channel limit
+                            </p>
+                          }
+                        >
+                          Max events per second
+                        </FormSectionLabel>
+                      }
+                    >
+                      <FormSectionContent loading={isLoading}>
+                        <FormControl_Shadcn_>
+                          <Input_Shadcn_
+                            {...field}
+                            type="number"
+                            disabled={!canUpdateConfig}
+                            value={field.value || ''}
+                            {...form.register('max_events_per_second', { valueAsNumber: true })}
+                          />
+                        </FormControl_Shadcn_>
+                      </FormSectionContent>
+                    </FormSection>
+                  )}
+                />
+              </CardContent>
+              <CardContent>
+                <FormField_Shadcn_
+                  control={form.control}
+                  name="max_bytes_per_second"
+                  render={({ field }) => {
+                    const { value, unit } = convertFromBytes(max_bytes_per_second ?? 0)
```

Comment:
you can use `field.value` here instead of `max_bytes_per_second`. The current approach with `form.watch` will cause entire form to rerender.

Reply:
oooh good catch 🙏